checkov/common/runners/base_runner.py

```diff
@@ -12,11 +12,36 @@
 
 class BaseRunner(ABC):
     check_type = ""
+    definitions = None
+    context = None
+    breadcrumbs = None
+    external_registries = None
+    graph_manager = None
+    graph_registry = None
 
     @abstractmethod
     def run(self, root_folder, external_checks_dir=None, files=None, runner_filter=RunnerFilter(), collect_skip_comments=True):
         pass
 
+    def set_external_data(self, definitions: Optional[Dict], context: Optional[Dict], breadcrumbs: Optional[Dict]):
+        self.definitions = definitions
+        self.context = context
+        self.breadcrumbs = breadcrumbs
+
+    def load_external_checks(self, external_checks_dir: List[str]):
+        pass
+
+    def get_graph_checks_report(self, root_folder, runner_filter: RunnerFilter):
+        pass
```

Comment:
should these be abstract methods?

Reply:
No, otherwise I'll have to force every runner to implement them, even if they don't need them yet. This way each runner can "choose" whether to implement them or not.